source/server/listener_impl.cc

```diff
@@ -678,10 +678,24 @@ void ListenerImpl::buildSocketOptions() {
     connection_balancer_ = std::make_shared<Network::ExactConnectionBalancerImpl>();
 #else
     // Not in place listener update.
-    if (config_.has_connection_balance_config()) {
-      // Currently exact balance is the only supported type and there are no options.
-      ASSERT(config_.connection_balance_config().has_exact_balance());
-      connection_balancer_ = std::make_shared<Network::ExactConnectionBalancerImpl>();
+    if (config_.has_connection_balance_config() &&
```

Comment:
You can use a switch statement here.

Reply:
OK.

Reply:
Double-check the API, only exact or extend connection balancer, so simplify it.